### PR TITLE
Core: Add catalog type for glue,jdbc,nessie

### PR DESCRIPTION
**Pull request by ajantha-bhat**

### CODE REVIEW

File: docs/docs/aws.md [hunk 1]
```diff
@@ -428,7 +428,7 @@ For example, to add S3 delete tags with Spark 3.3, you can start the Spark SQL s
 ```
 sh spark-sql --conf spark.sql.catalog.my_catalog=org.apache.iceberg.spark.SparkCatalog \
     --conf spark.sql.catalog.my_catalog.warehouse=s3://iceberg-warehouse/s3-tagging \
-    --conf spark.sql.catalog.my_catalog.catalog-impl=org.apache.iceberg.aws.glue.GlueCatalog \
+    --conf spark.sql.catalog.my_catalog.type=glue \
```
Comment (nastra) [inline docs/docs/aws.md:431]: there are more examples of `catalog-impl` in this file that should be adjusted

Comment (ajantha-bhat) [reply docs/docs/aws.md:431]: I intentionally left it thinking examples can have both the kinds.

Let me update it then.

Comment (ajantha-bhat) [reply docs/docs/aws.md:432]: updated and handled some more places + config docs.

File: docs/docs/flink-configuration.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ The following properties can be set globally and are not limited to a specific c
...
 | ---------------------------- |----------| -------------------------- |----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | type                         | ✔️       | iceberg                    | Must be `iceberg`.                                                                                                                                                               |
-| catalog-type                 |          | `hive`, `hadoop` or `rest` | `hive`, `hadoop` or `rest` for built-in catalogs, or left unset for custom catalog implementations using catalog-impl.                                                           |
+| catalog-type                 |          | `hive`, `hadoop`, `rest`, `glue`, `jdbc` or `nessie` | The underlying Iceberg catalog implementation, `HiveCatalog`, `HadoopCatalog`, `RESTCatalog`, `GlueCatalog`, `JdbcCatalog`, `NessieCatalog` or left unset if using a custom catalog|
```
Comment (nastra) [inline docs/docs/flink-configuration.md:40]:
```suggestion
| catalog-type                 |          | `hive`, `hadoop`, `rest`, `glue`, `jdbc` or `nessie` | The underlying Iceberg catalog implementation, `HiveCatalog`, `HadoopCatalog`, `RESTCatalog`, `GlueCatalog`, `JdbcCatalog`, `NessieCatalog` or left unset if using a custom catalog implementation via catalog-impl|
```

**Pull request merged**